torch/csrc/jit/script/compiler.cpp

```diff
@@ -1396,17 +1396,77 @@ struct to_ir {
       const Ident& target,
       const List<Expr>& args,
       const List<Stmt>& body) {
-    // TODO: start, stop, step loop
-    if (args.size() != 1) {
+    Value *end_val = nullptr, *start_val = nullptr, *step_val = nullptr;
+    bool isSimpleRange = (args.size() == 1);
+    std::vector<Value*> argVals;
+    for (auto i: args) {
+      argVals.push_back(ensureInt(range, emitExpr(i)));
+    }
+    if (isSimpleRange) {
+      end_val = argVals[0];
+      start_val = end_val->owningGraph()->insertConstant(0);
+      step_val = end_val->owningGraph()->insertConstant(1);
+    } else if (args.size() == 2 || args.size() == 3) {
+      start_val = argVals[0];
+      end_val = argVals[1];
+      if (args.size() == 3) {
+        step_val = argVals[2];
+      } else {
+        step_val = end_val->owningGraph()->insertConstant(1);
+      }
+    } else if (args.size() == 0) {
+      throw ErrorReport(range)
+          << "range expected 1 arguments, got 0";
```

Comment:
Why do we need this branch? seems like the `else` below would be sufficient.

Reply:
This is the behavior in Python.
```python
>>> range()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: range expected 1 arguments, got 0
>>> range(1,2,3,4)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: range expected at most 3 arguments, got 4
```

Seems to make more sense to me to output an error like `range expected 1-3 arguments, got x`, but that's Python's behavior 🤷‍♂

Reply:
lol ok whatever